src/rgw/driver/motr/rgw_sal_motr.h

```diff
@@ -1141,6 +1141,7 @@ class MotrStore : public StoreDriver {
 
 struct obj_time_weight {
   real_time mtime;
+  real_time internal_mtime;
```

Comment:
can we just decode `RGW_ATTR_INTERNAL_MTIME` into `mtime` when it exists? or is there a specific need to track it separately here?

Reply:
@cbodley I am not quite sure, if INTERNAL_MTIME is set then mtime struct will pick this value rather than mtime. If it is actual multisite scenario where the object is modified after cloud restore. Then mtime is modified but internal mtime is the same. If we map internal mtime to this structs mtime then sync may not happen, right?

Reply:
> If it is actual multisite scenario where the object is modified after cloud restore.

there are a couple cases where we artificially increment the mtime to force replication (`RadosObject::set_obj_attrs()` and `resync_encrypted_multipart()`) that should be using this new INTERNAL_MTIME strategy instead

in other cases like PutObject, we're overwriting the head object so any attrs like RGW_ATTR_INTERNAL_MTIME would disappear

Reply:
@cbodley i think the idea was to make this generic to any operation that needs to sync without needing an `mtime` modification and to avoid artificially forced replication. decoding `RGW_ATTR_INTERNAL_MTIME` into `mtime` at source would mean altering the `mtime`

Reply:
> decoding `RGW_ATTR_INTERNAL_MTIME` into `mtime` at source would mean altering the `mtime`

the suggestion was to decode into `obj_time_weight::mtime` which should only alter these mtime comparisons

if RGW_ATTR_INTERNAL_MTIME is present, i don't think these comparisons should consider the original mtime at all

Reply:
that sounds good to me.